IdentityCore/tests/automation/ui_tests_lib/lab_api/MSIDAutomationBaseApiRequest.m

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDAutomationBaseApiRequest.h"
+
+@implementation MSIDAutomationBaseApiRequest
+
+#pragma mark - NSCopying
+
+- (nonnull id)copyWithZone:(nullable NSZone *)zone
+{
+    MSIDAutomationBaseApiRequest *request = [[MSIDAutomationBaseApiRequest allocWithZone:zone] init];
+    return request;
+}
+
+#pragma mark - MSIDTestAutomationRequest
+
+- (NSURL *)requestURLWithAPIPath:(NSString *)apiPath labAccessPassword:(NSString *)labPassword
+{
+    NSString *requestOperationPath = [self requestOperationPath];
+    
+    if (!requestOperationPath)
+    {
+        return nil;
+    }
+    
+    NSString *fullAPIPath = [apiPath stringByAppendingPathComponent:requestOperationPath];
+    NSURLComponents *components = [[NSURLComponents alloc] initWithString:fullAPIPath];
+    
+    NSMutableArray *queryItems = [NSMutableArray array];
+    [queryItems addObject:[[NSURLQueryItem alloc] initWithName:@"code" value:labPassword]];
+    
+    NSArray *extraQueryItems = [self queryItems];
+    
+    if (!extraQueryItems)
```

Comment:
Interesting, I see how current APIs always need some type of query items, but will this be viable for other potential calls?

Reply:
We can remove it in the future without any impact on other code if we see the case. For now, would prefer to start strict.